pkg/agent/nodeportlocal/k8s/annotations.go

```diff
@@ -45,106 +46,79 @@ func toJSON(serialize interface{}) string {
-	if current == nil {
-		current = make(map[string]string)
-	}
-
-	klog.V(2).Infof("Building annotation for Pod: %s\tport: %v --> %v:%v", pod.Name, containerPort, nodeIP, nodePort)
-
-	var annotations []NPLAnnotation
-	if current[NPLAnnotationKey] != "" {
-		if err = json.Unmarshal([]byte(current[NPLAnnotationKey]), &annotations); err != nil {
-			klog.Warningf("Unable to unmarshal NodePortLocal annotation: %v", current[NPLAnnotationKey])
+// IsNPLAnnotationRequired returns true if a new NPL Annotation value is required. It checks
+// for the Container Port, Node Port and the Pod IP in the existing list of the NPL Annotations
+// of the Pod.
+func IsNPLAnnotationRequired(annotations map[string]string, nodeIP string, containerPort, nodePort int) bool {
+	var nplAnnotations []NPLAnnotation
+	if annotations[NPLAnnotationKey] != "" {
+		if err := json.Unmarshal([]byte(annotations[NPLAnnotationKey]), &nplAnnotations); err != nil {
+			klog.Warningf("Unable to unmarshal NodePortLocal annotation: %v", annotations[NPLAnnotationKey])
 		}
-
-		if !isNodePortInAnnotation(annotations, nodePort) {
-			annotations = append(annotations, NPLAnnotation{
-				PodPort:  containerPort,
-				NodeIP:   nodeIP,
-				NodePort: nodePort,
-			})
-		}
-	} else {
-		annotations = []NPLAnnotation{{
-			PodPort:  containerPort,
-			NodeIP:   nodeIP,
-			NodePort: nodePort,
-		}}
 	}
-
-	current[NPLAnnotationKey] = toJSON(annotations)
-	pod.Annotations = current
-}
-
-func removePodAnnotation(pod *corev1.Pod) {
-	klog.V(2).Infof("Removing entire NodePortLocal annotation from Pod: %s/%s", pod.Namespace, pod.Name)
-	delete(pod.Annotations, NPLAnnotationKey)
-}
-
-func removeFromPodAnnotation(pod *corev1.Pod, containerPort int) {
-	var err error
-	current := pod.Annotations
-
-	klog.V(2).Infof("Removing NodePortLocal annotation from Pod: %s/%s\tport: %v", pod.Namespace, pod.Name, containerPort)
-	var annotations []NPLAnnotation
-	if err = json.Unmarshal([]byte(current[NPLAnnotationKey]), &annotations); err != nil {
-		klog.Warningf("Unable to unmarshal NodePortLocal annotation: %v", current[NPLAnnotationKey])
-		return
+	if isNodePortInAnnotation(nplAnnotations, nodePort, containerPort) {
+		// no updates required to the Pod
+		return false
 	}
+	return true
+}
 
+func removeFromNPLAnnotation(annotations []NPLAnnotation, containerPort int) []NPLAnnotation {
 	for i, ann := range annotations {
 		if ann.PodPort == containerPort {
 			annotations = append(annotations[:i], annotations[i+1:]...)
 			break
 		}
 	}
+	return annotations
+}
 
-	current[NPLAnnotationKey] = toJSON(annotations)
-	pod.Annotations = current
+func (c *NPLController) updatePodNPLAnnotation(pod *corev1.Pod, annotations []NPLAnnotation) error {
+	if err := patchPod(annotations, pod, c.kubeClient); err != nil {
+		klog.Warningf("Unable to patch NPL annotations for Pod %s/%s: %s", pod.Namespace, pod.Name, err.Error())
+	}
+	klog.V(2).Infof("Successfully updated annotation for Pod %s/%s", pod.Namespace, pod.Name)
```

Comment:
logs & error messages should be consistent:
* "NPL annotations" vs "annotation"
* sometimes annotation is capitalized, sometimes it's not
* sometimes you spell out NodePortLocal, sometimes you use NPL

Reply:
Fixed it and made it consistent for the respective files. Thanks.